hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/UserMetricsContext.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.metrics;
+
+import com.hazelcast.internal.metrics.LongProbeFunction;
+import com.hazelcast.internal.metrics.ProbeBuilder;
+import com.hazelcast.internal.metrics.ProbeLevel;
+import com.hazelcast.internal.metrics.ProbeUnit;
+import com.hazelcast.jet.core.metrics.Counter;
+import com.hazelcast.jet.impl.execution.ProcessorTasklet;
+import com.hazelcast.jet.impl.util.Util;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+public class UserMetricsContext {
+
+    private final Function<String, Counter> counterSupplier;
+
+    private String onlyMetric;
+    private Counter onlyCounter;
+    private Map<String, Counter> metrics;
+
+    public UserMetricsContext(ProbeBuilder probeBuilder, ProcessorTasklet source, ProbeLevel level, ProbeUnit unit) {
+        this.counterSupplier = metric -> {
```

Comment:
Why do we need a supplier? It's used only once.

Reply:
Will inline.